app/controllers/admin/enterprises_controller.rb

```diff
@@ -311,5 +311,20 @@ def location_after_save
     def ams_prefix_whitelist
       [:index, :basic]
     end
+
+    def enterprise_params
+      return params[:enterprise] if params[:enterprise].empty?
+
+      params.require(:enterprise).permit(
+        :name, :is_primary_producer, :visible, :permalink,
+        :contact_name, :email_address, :phone, :sells, :owner_id,
+        :website, :facebook, :instagram, :linkedin, :twitter,
+        :abn, :acn, :charges_sales_tax, :display_invoice_logo,
+        :invoice_text, :description, :long_description, :promo_image,
+        :preferred_product_selection_from_inventory_only, :preferred_shopfront_message,
+        :preferred_shopfront_closed_message, :preferred_shopfront_taxon_order,
+        :preferred_shopfront_order_cycle_order, :require_login,
+        :allow_guest_orders, :allow_order_changes, :enable_subscriptions)
```

Comment:
If we change this to include:
```ruby
...
   :allow_guest_orders, :allow_order_changes, :enable_subscriptions,
   producer_properties_attributes: [:id, :property_name, :value, :_destroy])
end
```

...we fix another 6+ specs related to saving producer properties. I can submit it separately after this is merged, if it's easier.

Also in the above example, `producer_properties_attributes` is one of those annoying arrays of unknown length where we need to permit attributes on each item in the array. With this array syntax it apparently **Just Works™**...

Reply:
in what spec? I only see one fixed in admin/enterprises_controller

Reply:
Producer properties tests in `spec/features/admin/enterprises_spec.rb`.

Reply:
I've just noticed we also need `user_ids: []` in the list as well.

Sorry for the constant stream of annoying additions. :sweat_smile:

Reply:
done :+1: